content/deyemiobaa/building-a-todolist-with-rails.md

```diff
@@ -0,0 +1,354 @@
+| PATCH/PUT | /todos/:id | todos#update | update a specific todo |
+| DELETE    | /todos/:id | todos#destroy | delete a specific todo |
+
+## Creating a new todo item
+
+Now that we have our routes setup, we need add the corresponding method in our controller with some logic to handle the form submission. Open the `app/controllers/todos_controller.rb` file and replace what you have with the following code.
+
+```ruby
+class TodosController < ApplicationController
+  def index
+  end
+
+  def create
+    @todo = Todo.create(description: params[:todo][:description])
+    if @todo.valid?
+      redirect_to todos_path 
+    end
+  end
+end
+```
+
+The `create` method creates a new instance of the Todo model and saves it to the database. The `description` attribute is set to the value of the `description` field in the form. The `params` method returns a hash of the form data. The hash looks like this.
+
+```ruby
+{
+  todo: {
+    description: "new todo"
+  }
+}
+```
+
+The `params[:todo][:description]` syntax gets the value of the hash. The `valid?` method checks if the model is valid. If the model is valid it renders the `index` view and show the list of todos. If the model is not valid, it just renders the `index` view. In a real application, we would want to display the errors to the user.
+
+You can submit the form multiple time to have enough data to test the table. We'll add the table in the next section. You can verify that your todos are being saved to the database by opening your terminal to your apps directory and running `rails c` to open the rails console. Then run `Todo.all` to see the list of todos.
+
+## Displaying the list of todos
+
+In our view, we need to add a table to display the list of todos. Open the `app/views/todos/index.html.erb` file and append the following code to it.
+
+```erb
+...
+<table class="w-3/5 mt-10 border-separate table-auto text-slate-500">
+  <thead>
+    <tr>
+      <th class="p-3 text-sm font-semibold text-left text-gray-700 bg-gray-200">ID</th>
+      <th class="p-3 text-sm font-semibold text-left text-gray-700 bg-gray-200">Description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <% @todos.each do |todo| %>
+      <tr>
+        <td class="px-3 py-4 text-sm font-normal text-left text-gray-600 bg-gray-50">
+          <%= todo.id %>
+        </td>
+        <td class="px-3 py-4 text-sm font-normal text-left text-gray-600 bg-gray-50">
+          <%= todo.description %>
+        </td>
+      </tr>
+    <% end %>
+  </tbody>
+</table>
+```
+
+This code will render a table with the list of todos. The `@todos` variable is an array of Todo models. The `each` method iterates over the array and renders a row for each todo. The `todo.id` and `todo.description` syntax gets the id and description of the todo.
+
+In order to display the list of todos, we need to get the list of todos from the database. Open the `app/controllers/todos_controller.rb` and add the following code to the `index` method.
```

Comment:
In the previous section, we started with the controller and then went on to the view. Can we follow this same format for the entirety of the tutorial?